website/static/js/share/results.js

```diff
@@ -37,79 +38,167 @@ Results.controller = function(vm) {
+                onclick: function() {
+                    utils.updateFilter(self.vm, 'match:contributors.familyName:' + person.familyName, true);
+                    utils.updateFilter(self.vm, 'match:contributors.givenName:' + person.givenName, true);
+                }
+            }, person.name)
+        ]);
+    };
+
+    /* Renders all the contributors for a given result */
+    self.renderContributors = function(result) {
+        var renderPeople = function(people) {
+            return $.map(people, self.renderPerson);
+        };
+
+        return m('span.pull-left', {style: {'text-align': 'left'}},
+            result.showAllContrib || result.contributors.length < 8 ?
+                renderPeople(result.contributors) :
+                m('span', [
+                    renderPeople(result.contributors.slice(0, 7)),
+                    m('br'),
+                    m('a', {onclick: function(){result.showAllContrib = result.showAllContrib ? false : true;}}, 'See All')
+                ])
+        );
+    };
+
+    /* Renders the subjects (sort of like tags) for a result */
+    self.renderSubjects = function(result) {
+        var rendersubject = function(subject) {
+            return [
+                m('.badge.pointer', {onclick: function(){
+                    utils.updateFilter(self.vm, 'match:subjects:"' + subject + '"', true);
+                }}, subject.length < 50 ? subject : subject.substring(0, 47) + '...'),
+                ' '
+            ];
+        };
+
+        if (result.showAllsubjects || (result.subjects || []).length < 5) {
+            return $.map((result.subjects || []), rendersubject);
+        }
+        return m('span', [
+            $.map((result.subjects || []).slice(0, 5), rendersubject),
+            m('br'),
+            m('div', m('a', {onclick: function() {result.showAllsubjects = result.showAllsubjects ? false : true;}},'See All'))
+        ]);
+    };
+    
+    /* Renders the footer of a result, including release date, provider information and the raw/normalized records */
+    self.renderResultFooter = function(result) {
+        return m('div', [
+            m('span',
+                'Released on ' + new $osf.FormattableDate(result.providerUpdatedDateTime).local,
+                vm.rawNormedLoaded() ?  m('span', [
+                    m('span', {style: {'margin-right': '5px', 'margin-left': '5px'}}, ' | '),
+                    m('a', {
+                        onclick: function() {
+                            result.showRawNormed = result.showRawNormed ? false : true;
+                            if (!result.raw) {
+                                utils.loadRawNormalized(result);
+                            }
+                        }
+                    }, 'Data')
+                ]) : []
+            ),
+            m('span.pull-right', [
+                m('img', {src: self.vm.ProviderMap[result.shareProperties.source].favicon, style: {width: '16px', height: '16px'}}),
```

Comment:
For accessibility, all `img` tags should have `alt` text. http://a11yproject.com/posts/alt-text/